src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,76 @@
-import React from 'react';
+import React, { useState, Dispatch, SetStateAction } from 'react';
+import classNames from 'classnames';
+import { client } from '../utils/fetchClient';
+import { Post } from '../types/Post';
+
+interface T {
+  selectedPost: Post;
+  setIsLoading: Dispatch<SetStateAction<boolean>>;
+  isLoading: boolean;
+}
+
+export const NewCommentForm: React.FC<T> = ({
+  selectedPost,
+  setIsLoading,
+  isLoading,
+}) => {
+  const [inputName, setInputName] = useState('');
+  const [inputEmail, setInputEmail] = useState('');
+  const [inputText, setInputText] = useState('');
+  const [inputNameSend, setInputNameSend] = useState(false);
+  const [inputEmailSend, setInputEmailSend] = useState(false);
+  const [inputTextSend, setInputTextSend] = useState(false);
+
+  const onSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setIsLoading(true);
+    setInputNameSend(true);
+    setInputEmailSend(true);
+    setInputTextSend(true);
+
+    try {
+      const newComment = {
+        postId: selectedPost.id,
+        name: inputName,
+        email: inputEmail,
+        body: inputText,
+      };
+
+      if (inputName !== '' && inputEmail !== '' && inputText !== '') {
+        await client.post('/comments', newComment);
+      }
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const handleInputName = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputName(event.target.value);
+    setInputNameSend(false);
+  };
+
+  const handleInputEmail = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputEmail(event.target.value);
+    setInputEmailSend(false);
+  };
+
+  const handleInputText = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setInputText(event.target.value);
+    setInputTextSend(false);
+  };
```

Comment:
please don't repeat yourself. combine state into one object and change it with one function